packages/data/lib/src/config_model/config_data.dart

```diff
@@ -0,0 +1,28 @@
+const _prod = 'prod';
+
+class DataConfig {
+  final String baseUrl;
+  final String flavor;
+
+  const DataConfig(this.baseUrl, this.flavor);
+
+  factory DataConfig.fromJson(
+    Map<String, dynamic> json,
+    String flavor,
+  ) {
+    switch (flavor) {
+      case _prod:
+        return DataConfig(
+          json['prodUrl'],
+          flavor,
+        );
+      default:
+        return DataConfig(
+          json['sandBoxUrl'],
+          flavor,
+        );
+    }
```

Comment:
after u'll change into 2 files, there won't be any need in passing flavor to fromJson